java/main/com/google/privacy/differentialprivacy/LaplaceNoise.java

```diff
@@ -149,17 +150,58 @@ public ConfidenceInterval computeConfidenceInterval(
       long lInfSensitivity,
       double epsilon,
       @Nullable Double delta,
-      double confidenceLevel) {
-    // TODO: Implement confidence interval computation.
-    throw new UnsupportedOperationException("Not implemented yet.");
+      double alpha) {
+    checkConfidenceParameters(l0Sensitivity, lInfSensitivity, epsilon, delta, alpha);
+    ConfidenceInterval confInt =
+        computeConfidenceInterval(
+            noisedX, Noise.getL1Sensitivity(l0Sensitivity, lInfSensitivity) / epsilon, alpha);
+    return ConfidenceInterval.create(
+        Math.round(confInt.lowerBound()), Math.round(confInt.upperBound()));
   }
 
   private void checkParameters(double l1Sensitivity, double epsilon, @Nullable Double delta) {
-    DpPreconditions.checkEpsilon(epsilon);
+    DpPreconditions.checkEpsilonVeryStrict(epsilon);
     DpPreconditions.checkNoiseDelta(delta, this);
     DpPreconditions.checkL1Sensitivity(l1Sensitivity);
   }
 
+  private void checkConfidenceParameters(
+      int l0Sensitivity,
+      double lInfSensitivity,
+      double epsilon,
+      @Nullable Double delta,
+      double alpha) {
+    DpPreconditions.checkEpsilonStrict(epsilon);
+    DpPreconditions.checkNoiseDelta(delta, this);
+    DpPreconditions.checkSensitivities(l0Sensitivity, lInfSensitivity);
+    DpPreconditions.checkAlpha(alpha);
+  }
+
+  /**
+   * Computes a confidence interval that contains the raw value {@code x} passed to {@link
+   * #addNoise(double, int, double, double, Double)} with a probability equal to {@code 1 - alpha}
+   * based on the specified {@code noisedX} and lambda.
+   */
+  private ConfidenceInterval computeConfidenceInterval(
+      double noisedX, double lambda, double alpha) {
+    // Finding a symmetrical confidence interval around a Laplace of (0, lambda)
+    // by calculating Z_(alpha/2) using inverseCdfLaplace which will return a
+    // negative value by symmetry to gain more accuracy for extremely small alpha.
+    double z = inverseCdfLaplace(lambda, alpha / 2);
+    return ConfidenceInterval.create(noisedX + z, noisedX - z);
+  }
+
+  /**
+   * Computes the quantile z satisfying Pr[Y <= z] = p for a random variable Y that is Laplace
+   * distributed with the specified lambda where mean is zero.
+   */
+  private double inverseCdfLaplace(double lambda, double p) {
```

Comment:
as I mentioned for Gaussian noise, this can be static and consider naming the method as an action, e.g., "computeLaplacePercentile(...)"

Reply:
Most of the functions that are in this file could also be static, so I think having this non-static to follow the trend is better because it will be the only static function in the file and since its private, I don't think there is enough reason to change it to static. Is there is a reason to why it should be static?

For function name, method as an action is better for Java. I will change it. Thanks!